src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
     /// <exception cref="T:ArgumentNullException">Thrown when either of the input sequences is null.</exception>
     val prependSeq: source1: seq<'T> -> source2: taskSeq<'T> -> taskSeq<'T>
 
-    /// Returns taskSeq as an array. This function is blocking until the sequence is exhausted and will properly dispose of the resources.
+    /// <summary>
+    /// Builds an F# <see cref="list" /> from the input task sequence in <paramref name="source" />.
+    /// This function is blocking until the sequence is exhausted and will then properly dispose of the resources.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting list.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toList: source: taskSeq<'T> -> 'T list
 
-    /// Returns taskSeq as an array. This function is blocking until the sequence is exhausted and will properly dispose of the resources.
+    /// <summary>
+    /// Builds an <see cref="array" /> from the input task sequence in <paramref name="source" />.
+    /// This function is blocking until the sequence is exhausted and will then properly dispose of the resources.
+    /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting array.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val toArray: source: taskSeq<'T> -> 'T[]
 
     /// <summary>
-    /// Returns the task sequence <paramref name="source" /> as an F# <see cref="seq" />, that is, an
-    /// <see cref="IEnumerable&lt;'T>" />. This function is blocking at each <see cref="yield" />, but otherwise
-    /// acts as a normal delay-executed sequence.
-    /// It will then dispose of the resources.
+    /// Views the task sequence in <paramref name="source" /> as an F# <see cref="seq" />, that is, an
+    /// <see cref="IEnumerable&lt;'T>" />. This function is blocking at each <see cref="yield" /> or call
+    /// to <see cref="IEnumerable&lt;'T>/MoveNext()" /> in the resulting sequence.
+    /// Resources are disposed when the sequence is disposed, or the sequence is exhausted.
     /// </summary>
     ///
     /// <param name="source">The input task sequence.</param>
     /// <returns>The resulting task sequence.</returns>
```

Comment:
Maybe this could/should be reworded to convey that it does not *do* anything immediately, instead preparing a mapping overlay/view? "the resulting" might suggest it does something to someone that's not paying full attention and/or does not have the distinction between an enumerator and an iterator in mind?

Reply:
This uses the same wording as `seq` in F#, using the wording `Views the sequence as ...`. Same as with `returns`. I don't mind improving over F#'s own words, though, but we do say that we return a `task sequence`, which is always lazily evaluated.

I might deprecate the blocking methods, though, they are dangerous.